### DISCUSSION TITLE
Providing an access point for latest results - results.json

### DISCUSSION THREAD

**Discussion by XHalamaX** [Akkudoktor-EOS/EOS#286] [Ideas]
Hi everyone! 👋

I’m running into some challenges while testing EOS, and I’d really appreciate your insights and suggestions. Here’s what’s happening:

### 🐢 Issue: VPS Slower or Timeout Problems

It seems like the VPS might be running slower than in previous weeks, or perhaps something has changed. I’m using Node-RED and consistently getting a timeout after 120,000 ms.

I’ve tried increasing the timeout by setting:

`msg.httpRequestTimeout = 240000;`

…but unfortunately, it didn’t work.
### 🔍 Observations

Here’s what I’ve noticed:

- The Python process on the server console takes just a few seconds longer to complete.
- The visualization file (e.g., visualization_results.pdf) is generated slightly later, and I can confirm the creation time in the document details.

### 💡 Feature Request: Additional Results File

It would be fantastic if we could also get a results.json file in addition to the PDF file (visualization_results.pdf). This would make my testing much easier.
### 🛠️ Testing Suggestion

To improve testing, it would also be helpful to see Timestamps indicating when both the JSON and PDF results were created (so in the files itself)

Thanks so much in advance for your help! 🙏 If anyone has ideas about increasing the timeout or getting the results.json file, I’d be very grateful.

**Comment by NormannK** [comment]
Thanks for testing everything. Yesterday we merged a update which does produce a new pdf which is now named visualization_results_new.pdf. I'm hindsight we should change that back to the old name but the pdf should look way better. 
The other reason I could think of are some recent changes in the calculation which do take a little bit longer (depending on hardware oc) but fixed some edge cases but also have better results. 
Regarding the timestamp in th pdf: you are right recently we talked about that too. After the holidays I'll add the date and some raw output to the end of the file

**Reply by XHalamaX** [reply]
But could you also add an API to get the latest result in JSON format?  Not sure about the effort but then we don't care about timeouts and can just request the data at some time. I don't want to extract the data from the pdf 

This could help to run EOS on low performance systems.

Something like

GET /optimize
Or
GET /result

would be great.

Thank you very much

**Reply by XHalamaX** [reply]
@NormannK I can proceed with my setup and testing now. I found a solution for me for now but I think the request is maybe still valid.

I figured out that the timeout was comming from Node-RED and can be adjusted in the config file (settings.js). I changed
`httpRequestTimeout: 120000,`
to
`httpRequestTimeout: 240000,`

The alternative is decreasing of the optimization_hours in the EOS configuration file
`"optimization_hours": 48,`
to
`"optimization_hours": 24,`
